superduperdb/ext/llm/vllm.py

```diff
@@ -62,6 +63,47 @@ def build_post_data(self, prompt: str, **kwargs: dict[str, Any]) -> dict[str, An
         return {"prompt": prompt, **total_kwargs}
 
 
+class _VllmCore:
+    def __init__(self, **kwargs) -> None:
+        # Use kwargs to avoid incompatibility after vllm version upgrade
+        if "disable_log_stats" not in kwargs:
+            kwargs["disable_log_stats"] = True
+        from vllm.engine.arg_utils import AsyncEngineArgs
+        from vllm.engine.async_llm_engine import AsyncLLMEngine
+
+        kwargs.setdefault("disable_log_requests", True)
+        engine_args = AsyncEngineArgs(**kwargs)
+        self.engine = AsyncLLMEngine.from_engine_args(engine_args)
+
+    async def agenerate(self, prompt, **kwargs):
+        from vllm import SamplingParams
+        from vllm.utils import random_uuid
+
+        sampling_params = SamplingParams(**kwargs)
+        request_id = random_uuid()
+        results_generator = self.engine.generate(prompt, sampling_params, request_id)
```

Comment:
I may be wrong 
await ?
@jieguangzhou

Reply:
we use `async for` below

```
async for request_output in results_generator:
             final_output = request_output
```

This uses the same logic as the official asynchronous API call